static/app/views/alerts/rules/metric/insightsMetricField.tsx

```diff
@@ -159,8 +193,8 @@ function InsightsMetricField({aggregate, project, onChange}: Props) {
   );
 
   // When using the async variant of SelectControl, we need to pass in an option object instead of just the value
-  const selectedMriOption = selectedValues?.mri && {
-    label: formatMRI(selectedValues.mri),
+  const selectedOption = selectedValues?.mri && {
+    label: isMRI(selectedValues.mri) ? formatMRI(selectedValues.mri) : selectedValues.mri,
```

Comment:
I'm curious why the `isMRI` check is necessary? Given that `selectedValues.mri` already has type `MRI` (which is the type predicate that `isMRI` is checking, and what `formatMRI` expects) 🤔

Reply:
Hey, good question! So `selectedValues` comes from a helper `parseField` function which just naively pulls out the argument(`mri`) using regex, and then casts the type as `MRI` regardless of whether it is actually true or not.

Insights functions like `http_response_rate(3)` have arguments that are not `MRI`s which is why we now have to run into this type check scenario.

I'm going to update to not use `parseField` so we don't make any assumptions on type!